src/components/DataDimension/FormulaItem.js

```diff
@@ -0,0 +1,197 @@
+        clearTimeout(clickTimeoutId)
+        const to = setTimeout(function () {
+            if (tagname !== 'INPUT') {
+                onClick(id)
+            } else {
+                inputRef.current && inputRef.current.focus()
+            }
+        }, maxMsBetweenClicks)
+        setClickTimeoutId(to)
+    }
+
+    const handleDoubleClick = () => {
+        clearTimeout(clickTimeoutId)
+        setClickTimeoutId(null)
+        onDoubleClick(id)
+    }
+
+    const handleChange = (e) => onChange({ index, value: e.target.value })
+
+    const getContent = () => {
+        if (type === TYPE_INPUT) {
+            return (
+                <>
+                    {DragHandleIcon}
+                    <span className="input-positioner">
+                        <span className="width-machine" aria-hidden="true">
+                            {value}
+                        </span>
+                        <input
+                            id={id}
+                            name={label}
+                            onChange={handleChange}
+                            value={value}
+                            type="number"
+                            ref={inputRef}
+                        />
+                    </span>
+                    <style jsx>{styles}</style>
+                </>
+            )
+        }
+
+        if (type === TYPE_DATAELEMENT) {
+            return (
+                <>
+                    {getIcon(DIMENSION_TYPE_DATA_ELEMENT)}
+                    <span className="data-element-label">{label}</span>
+                    <style jsx>{styles}</style>
+                </>
+            )
+        }
+
+        return (
+            <>
+                <span className="operator-label">{label}</span>
+                <style jsx>{styles}</style>
+            </>
+        )
+    }
+
+    return (
+        <>
+            <div
+                ref={setNodeRef}
+                {...attributes}
+                {...listeners}
+                className={isLast && 'last-item'}
+                style={style}
+            >
+                <div
+                    className={cx('formula-item', {
+                        inactive: !isDragging,
+                        insertBefore: insertPosition === BEFORE,
+                        insertAfter: insertPosition === AFTER,
+                        highlighted: isHighlighted,
+                    })}
+                    tabIndex={isLast ? 0 : -1}
+                    onClick={handleClick}
+                    onDoubleClick={handleDoubleClick}
+                >
+                    <div className="content">{getContent()}</div>
+                </div>
+            </div>
+            <style jsx>{styles}</style>
+        </>
+    )
+}
+
+FormulaItem.propTypes = {
+    onChange: PropTypes.func.isRequired,
+    onClick: PropTypes.func.isRequired,
+    onDoubleClick: PropTypes.func.isRequired,
+    hasFocus: PropTypes.bool,
+    id: PropTypes.string,
```

Comment:
`onDoubleClick` is required and on line 110 it's used with `id` (`onDoubleClick(id)`) so that would make `id` required too?

Reply:
Switching to index